packages/fiori-elements-writer/src/index.ts

```diff
@@ -42,18 +42,36 @@ async function generate<T extends {}>(basePath: string, data: FioriElementsApp<T
 
     await addOdataService(basePath, feApp.service, fs);
 
+    const coercedUI5Version = semVer.coerce(feApp.ui5?.version)!;
     const templateOptions: TemplateOptions = {
-        changesPreview: feApp.ui5?.version
-            ? semVer.lt(semVer.coerce(feApp.ui5?.version)!, changesPreviewToVersion)
-            : false,
+        changesPreview: feApp.ui5?.version ? semVer.lt(coercedUI5Version, changesPreviewToVersion) : false,
         changesLoader: feApp.service.version === OdataVersion.v2
     };
 
     // Add new files from templates e.g.
     const rootTemplatesPath = join(__dirname, '..', 'templates');
     // Add templates common to all template types
-    // Common files
-    const ignore = [feApp.appOptions?.typescript ? '**/*.js' : '**/*.ts'];
+    const jsIgnoreGlob = ['**/*.ts'];
+
+    let ignore = jsIgnoreGlob;
+    if (feApp.appOptions?.typescript === true) {
+        // isV2FETypesAvailable - Boolean to indicate if V2 Fiori Element types were available in the UI5 version
+        const isV2FETypesAvailable = feApp.ui5?.version ? semVer.gte(coercedUI5Version, '1.108.0') : false;
```

Comment:
You are using a magic string `1.108.0` however, you also declared a constant in the `ui5-application-writer` with `1.108.0` for the FE types available version. Shouldn't you refer to the same constant in both cases?

Reply:
Continued thinking about that one: having a variable `V2_FE_TYPES_AVAILABLE` in `ui5-application-writer` that is only required for FE is not ok.
Please rethink, how you could keep the FE specifics completely in the fe writer.

Reply:
Fixes applied in https://github.com/SAP/open-ux-tools/pull/845/commits/53dc7bccc45149585893580868b322ed87039676